examples/ransac_comparison_autoreject.py

```diff
@@ -0,0 +1,212 @@
+ransac_ar = ransac_ar.fit(epochs)
+print("--- %s seconds ---" % (perf_counter() - start_time))
+
+corr_ar = ransac_ar.corr_
+bad_by_ransac_ar = ransac_ar.bad_chs_
+
+# Check channels that go bad together by RANSAC
+print("autoreject bad chs:", bad_by_ransac_ar)
+assert set(bad_ch_names) == set(bad_by_ransac_ar)
+
+# %%
+# pyprep's RANSAC
+
+start_time = perf_counter()
+bad_by_ransac_pyprep, corr_pyprep = ransac_pyprep.find_bad_by_ransac(
+    data=raw._data.copy(),
+    sample_rate=raw.info["sfreq"],
+    complete_chn_labs=np.asarray(raw.info["ch_names"]),
+    chn_pos=raw._get_channel_positions(),
+    exclude=[],
+    n_samples=n_samples,
+    sample_prop=fraction_good,
+    corr_thresh=corr_thresh,
+    frac_bad=fraction_bad,
+    corr_window_secs=corr_window_secs,
+    channel_wise=False,
+    random_state=rng,
+)
+print("--- %s seconds ---" % (perf_counter() - start_time))
+
+# Check channels that go bad together by RANSAC
+print("pyprep bad chs:", bad_by_ransac_pyprep)
+assert set(bad_ch_names) == set(bad_by_ransac_pyprep)
+
+# %%
+# Now we test the algorithms on real EEG data.
+# Let's download some data for testing.
+data_paths = mne.datasets.eegbci.load_data(subject=4, runs=1, update_path=True)
+fname_test_file = data_paths[0]
+
+# %%
+# Load data and prepare it
+
+raw = mne.io.read_raw_edf(fname_test_file, preload=True)
+
+# The eegbci data has non-standard channel names. We need to rename them:
+mne.datasets.eegbci.standardize(raw)
+
+# Add a montage to the data
+montage_kind = "standard_1005"
+montage = mne.channels.make_standard_montage(montage_kind)
+raw.set_montage(montage)
+
+
+# %%
+# autoreject's RANSAC
+ransac_ar = Ransac(
+    picks=None,
+    n_resample=n_samples,
+    min_channels=fraction_good,
+    min_corr=corr_thresh,
+    unbroken_time=fraction_bad,
+    n_jobs=1,
+    random_state=rng,
+)
+epochs = mne.make_fixed_length_epochs(
+    raw,
+    duration=corr_window_secs,
+    preload=True,
+    reject_by_annotation=False,
+    verbose=None,
+)
+
+start_time = perf_counter()
+ransac_ar = ransac_ar.fit(epochs)
+print("--- %s seconds ---" % (perf_counter() - start_time))
+
+corr_ar = ransac_ar.corr_
+bad_by_ransac_ar = ransac_ar.bad_chs_
+
+# Check channels that go bad together by RANSAC
+print("autoreject bad chs:", bad_by_ransac_ar)
+
+
+# %%
+# pyprep's RANSAC
+
+start_time = perf_counter()
+bad_by_ransac_pyprep, corr_pyprep = ransac_pyprep.find_bad_by_ransac(
+    data=raw._data.copy(),
+    sample_rate=raw.info["sfreq"],
+    complete_chn_labs=np.asarray(raw.info["ch_names"]),
+    chn_pos=raw._get_channel_positions(),
+    exclude=[],
+    n_samples=n_samples,
+    sample_prop=fraction_good,
+    corr_thresh=corr_thresh,
+    frac_bad=fraction_bad,
+    corr_window_secs=corr_window_secs,
+    channel_wise=False,
+    random_state=rng,
+)
+print("--- %s seconds ---" % (perf_counter() - start_time))
+
+# Check channels that go bad together by RANSAC
+print("pyprep bad chs:", bad_by_ransac_pyprep)
```

Comment:
we should add one more code block below to easily show:

- chs bad that are equal between pyprep and AR
- chs bad only AR
- chs bad only pyprep